src/sipreg/reg.c

```diff
@@ -297,7 +297,10 @@ static int request(struct sipreg *reg, bool reset_ls)
 
 	if (reset_ls) {
 		sip_loopstate_reset(&reg->ls);
-		sip_auth_reset(reg->auth);
+		/* the dialog will stop in case of a de-registration so we
+		 * must skip the digest challenge */
+		if (!reg->terminated)
```

Comment:
Previous commit was mine d8cfc30e4cbfc8fc03fccaf58c7186e5d49e30b0. We checked again if a re-REGISTER should start with a new digest challenge and found that it might not be necessary. We found this examples:
- re-REGISTRATION: https://tools.ietf.org/html/rfc3665#section-2.2
- de-REGISTRATION: https://tools.ietf.org/html/rfc3665#section-2.4

I tested with https://routr.io/ and sip.linphone.org. Anyway, if the SIP registrar does not accept the old nonce, it will answer with a Status 401 and then we start a new digest challenge. 

So we think that line 300 (the call to sip_auth_reset()) should be completely removed.

Reply:
I wouldn't mind that. Should I update the PR?

Reply:
I would say yes. We suggest to remove the line completely.